File: example3.js
class K {
  constructor() {
    this.value = 0
  }

  increment() {
    this.value = this.value + 1
  }

  count() {
    return this.value
  }
}

k = new K()
k.increment()
k.increment()
k.increment()
console.log(k.count())

File: example3.exs
defmodule K do
  use GenServer

  def init(_) do
    {:ok, 0}
  end

  def handle_cast(:increment, count) do
    {:noreply, count + 1}
  end

  def handle_call(:count, _, count) do
    {:reply, count, count}
  end
end

{:ok, pid} = GenServer.start_link(K, [])
GenServer.cast(pid, :increment)
GenServer.cast(pid, :increment)
GenServer.cast(pid, :increment)
IO.inspect GenServer.call(pid, :count)
